.github/workflows/workflow.yml

```diff
@@ -0,0 +1,78 @@
+# Builds and performs various check
+
+---
+name: Build
+
+on: 
+  push:
+  pull_request:
+
+jobs:
+  test:
+    name: Test
+    strategy:
+      # Keep running so we can see if other tests pass
+      fail-fast: false
+      matrix:
+        python-version:
+          - '3.6'
+        os:
+          - ubuntu-20.04
+        commands: 
+          - 'upgrade'
+          - 'install'
+    runs-on: ${{ matrix.os }}
+    services:
+      postgres:
+        image: postgres:11
+        env:
+          POSTGRES_DB: omero        
+          POSTGRES_PASSWORD: omero
+          POSTGRES_USER: omero
+        ports:
+          - 5432:5432
+        # Set health checks to wait until postgres has started
+        options: >-
+          --health-cmd pg_isready
+          --health-interval 10s
+          --health-timeout 5s
+          --health-retries 5
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Set up JDK 11
+        uses: actions/setup-java@v1
+        with:
+          java-version: 11
+      - name: Install slice2java
```

Comment:
I again note that some of this would be better refactored either into a re-usable workflow or into a script somewhere.

Reply:
yes.If we are okay with the name of the repo I have put in place of re-usable workflows (https://github.com/jburel/action-workflows) I could start refactoring

Reply:
ome/action-workflows seems fine. Worst case we have to re-adjust in the future.

Reply:
Actually I think that in that case we will need composite action i.e. https://github.com/jburel/action-workflows/blob/main/action.yml
So the repository mentioned is not going to work since we will need one repo per composite action i.e. something similar to ansible roles.